File: leetcode/python/0226_invert_binary_tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        def helper(node):
            if node != None:
                helper(node.left)
                node.left, node.right = node.right, node.left
                helper(node.left)

        helper(root)
        return root

File: leetcode/cpp/0226_invert_binary_tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void invertTreeHelper(TreeNode* node) {
        if (node != nullptr) {
            invertTreeHelper(node->left);

            TreeNode* leftNode = node->left;
            node->left = node->right;
            node->right = leftNode;

            invertTreeHelper(node->left);
        }
    }

    TreeNode* invertTree(TreeNode* root) {
        invertTreeHelper(root);

        return root;
    }
};